csrc/python_frontend/segmentation.h

```diff
@@ -185,6 +185,43 @@ class SegmentationState {
       const std::unordered_map<const Val*, int64_t>& map_value_to_original_fid,
       const at::ArrayRef<c10::IValue>& inputs);
 
+  // Given an empty FusionDefinition and a segment id, buildSegment creates the
+  // CPP Fusion, translates it to the python FusionDefinition, then return a
+  // mapping from segment fusion state indices to the original fusion state
+  // indices.
+  //
+  // NOTE: Steps 4a through 4d are run for every fusion segment. However,
+  // sometimes the python definition needs the extents of the original fusion's
+  // input tensors as extra arguments. Steps 4f to 4l creates mappings for these
+  // missing extents.
+  //
+  // Details:
+  //  1) Use segment id to get SegmentedGroup from group_run_order_.
+  //  2) Create CPP Fusion for SegmentedGroup.
+  //  * IrCloner acts as a map from fusion segment to the original fusion.
+  //  3) Translate CPP Fusion to Python FusionDefinition
+  //  4) Create map from segment fusion indices to original fusion indices.
+  //    a) Get cloned Vals for SegmentedGroup's inputs and outputs. Map them
+  //       to their original fusion indices.
+  //    b) Map cloned Vals to their segment Vals
+  //    c) Map segment Vals to their fusion indices.
+  //    d) Map original indices to segment indices.
```

Comment:
As @jjsjann123 suggested below, this detail will be more helpful in the CPP file to follow the mapping.